registrations/permissions.py

```diff
@@ -1,46 +1,91 @@
-from django.utils.translation import gettext as _
 from rest_framework import permissions
 from rest_framework.request import Request
 from rest_framework.views import APIView
 
 from events.auth import ApiKeyUser
-from registrations.models import SignUp
+from events.permissions import UserDataFromRequestMixin
+from registrations.models import Registration, SignUp
 
 
-class CanCreateEditDeleteSignup(permissions.BasePermission):
-    message: str = _(
-        "Only authenticated users are able to access sign-ups. Viewing, editing and deleting are"
-        "allowed only for admins of the publishing organization and for users that have created "
-        "the sign-up."
-    )
-
+class CanAccessRegistration(UserDataFromRequestMixin, permissions.BasePermission):
     def has_permission(self, request: Request, view: APIView) -> bool:
-        return (
-            request.method
-            in permissions.SAFE_METHODS + ("POST", "PUT", "PATCH", "DELETE")
-            and request.user.is_authenticated
-        )
+        if not request.user.is_authenticated:
```

Comment:
Well, I've learned something new here. django-cors-headers middleware will NOT pass an OPTIONS request further down the chain, but instead returns a response immediately.

( https://github.com/adamchainz/django-cors-headers/blob/main/src/corsheaders/middleware.py#L71 )

So my earlier comment a few weeks ago about the OPTIONS was not accurate. 

So this will work just fine, although in principle it would block all OPTIONS requests should they ever arrive here since OPTIONS don't carry authentication information. But yeah, goes in the "this should never happen" category.